cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/dataset/BusinessMetadataDataset.java

```diff
@@ -129,23 +198,49 @@ public BusinessMetadataRecord getBusinessMetadata(Id.NamespacedId targetId, Stri
   }
 
   /**
+   * Retrieves all the properties for the specified {@link Id.NamespacedId}.
+   *
+   * @param targetId the {@link Id.NamespacedId} for which properties are to be retrieved
+   * @return the properties of the specified {@link Id.NamespacedId}
+   */
+  public Map<String, String> getProperties(Id.NamespacedId targetId) {
```

Comment:
I always like abstraction of class over String if possible. This will allow us to add more stuff when searching for metadata.